### PR TITLE
Add rebase action

### PR DESCRIPTION
**Pull request by szaimen**
Copied from https://github.com/nextcloud/talk-android/pull/1300
Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

File: workflow-templates/rebase.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+on:
+    issue_comment:
+        types: [ created ]
+name: Automatic Rebase
+jobs:
+    rebase:
+        name: Rebase
+        if: github.event.issue.pull_request != '' && contains(github.event.comment.body, '/rebase')
```
Comment (skjnldsv) [inline workflow-templates/rebase.yml:8]: There was a few safety checks I added.
```suggestion
    # On pull requests and if the comment starts with `/compile`
    if: github.event.issue.pull_request != '' && startsWith(github.event.comment.body, '/rebase')
```

**Review by skjnldsv — COMMENTED** [review]
Let's space this out and indent with 2 space like other yml :)

**Comment by skjnldsv** [comment]
Also related: https://github.com/skjnldsv/vueexample/blob/master/.github/workflows/command-compile.yml

(still in beta, seems to work fine)

**Comment by skjnldsv** [comment]
Maybe also rename it to `command-rebase.yml`?

File: workflow-templates/rebase.yml (hunk 1 above)
Comment (szaimen) [reply workflow-templates/rebase.yml]: done

File: workflow-templates/command-rebase.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+on:
```
Comment (skjnldsv) [inline workflow-templates/command-rebase.yml:1]:
```suggestion
# This workflow is provided via the organization template repository
#
# https://github.com/nextcloud/.github
# https://docs.github.com/en/actions/learn-github-actions/sharing-workflows-with-your-organization

on:
```

File: workflow-templates/command-rebase.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  issue_comment:
+    types: [ created ]
+name: Automatic Rebase
+jobs:
```
Comment (skjnldsv) [inline workflow-templates/command-rebase.yml:5]:
```suggestion
name: Automatic Rebase

jobs:
```

File: workflow-templates/command-rebase.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      -   name: Checkout the latest code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      -   name: Automatic Rebase
+        uses: cirrus-actions/rebase@1.3.1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (skjnldsv) [inline workflow-templates/command-rebase.yml:19]:
```suggestion
    runs-on: ubuntu-latest

    steps:
      - name: Checkout the latest code
        uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - name: Automatic Rebase
        uses: cirrus-actions/rebase@1.3.1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Review by skjnldsv — CHANGES_REQUESTED** [review]
You're missing a [properties](https://github.com/nextcloud/.github/blob/master/workflow-templates/node.properties.json) file as well as an icon (check docs)
https://docs.github.com/en/actions/learn-github-actions/sharing-workflows-with-your-organization

File: workflow-templates/command-rebase.yml (hunk 2 above)
Comment (szaimen) [reply workflow-templates/command-rebase.yml:6]: done

File: workflow-templates/command-rebase.yml (hunk 4 above)
Comment (szaimen) [reply workflow-templates/command-rebase.yml]: done

File: workflow-templates/command-rebase.yml (hunk 3 above)
Comment (szaimen) [reply workflow-templates/command-rebase.yml]: done

**Comment by szaimen** [comment]
> You're missing a [properties](https://github.com/nextcloud/.github/blob/master/workflow-templates/node.properties.json) file as well as an icon (check docs)

Done. The icon is from https://www.iconfinder.com/icons/298788/merge_git_icon (licensed under MIT)

**Comment by szaimen** [comment]
I've found a better on https://www.iconfinder.com/icons/298789/pull_git_request_icon

**Review by skjnldsv — APPROVED** [review]

File: workflow-templates/command-rebase.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the latest code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      
+      - name: Automatic Rebase
+        uses: cirrus-actions/rebase@1.3.1
```
Comment (skjnldsv) [inline workflow-templates/command-rebase.yml:25]: https://github.com/cirrus-actions/rebase/releases/tag/1.5

File: workflow-templates/command-rebase.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (skjnldsv) [inline workflow-templates/command-rebase.yml:27]: The only issue with that is that github-action does not trigger CI.
We might have to think a bit more about this (closing/opening the pr will trigger ci again)

Comment (szaimen) [reply workflow-templates/command-rebase.yml:27]: I found this: https://github.com/cirrus-actions/rebase
image

File: workflow-templates/command-rebase.yml (hunk 5 above)
Comment (szaimen) [reply workflow-templates/command-rebase.yml]: done

File: workflow-templates/command-rebase.yml (hunk 6 above)
Comment (szaimen) [reply workflow-templates/command-rebase.yml:27]: @skjnldsv what do you think?

**Comment by nickvergessen** [comment]
> and indent with 2 space like other yml :)

Time to add an `.editorconfig` with those 3 lines: https://github.com/nextcloud/server/blob/master/.editorconfig#L17-L19

**Comment by szaimen** [comment]
> Time to add an `.editorconfig` with those 3 lines: https://github.com/nextcloud/server/blob/master/.editorconfig#L17-L19

You mean to this repo?

**Comment by nickvergessen** [comment]
yes

File: workflow-templates/command-rebase.yml (hunk 6 above)
Comment (szaimen) [reply workflow-templates/command-rebase.yml:27]: @skjnldsv if we don't find a better solution in the short term, can we merge this and improve it later?

Comment (skjnldsv) [reply workflow-templates/command-rebase.yml:27]: This needs to be discussed with @LukasReschke :)
This requires simple standard write + workflow permissions:

Comment (skjnldsv) [reply workflow-templates/command-rebase.yml:27]: image

**Review by skjnldsv — APPROVED** [review]

File: workflow-templates/command-rebase.yml (hunk 6 above)
Comment (LukasReschke) [reply workflow-templates/command-rebase.yml:27]: I'm not a super big fan of adding a personal access token if we can avoid it. This usually means anyone in the org could access the token 🙈 (if we go that way, at least make it a dedicated service account in the organization)

That said: do we need "workflow permissions" even? I believe that is only needed to update a workflow file not to execute one?

I'd really like the inverse: The GitHub action would trigger a webhook and then some bot somewhere invokes the GitHub workflow with parameters using the REST API. (I believe one can parameterise this - which is what we also did on the Docker repository) Then at least one can't easily exfiltrate a token.

Comment (skjnldsv) [reply workflow-templates/command-rebase.yml:27]: The issue is having to run the bot and maintain it. They're usually not that easy and this is what we've been doing. But also what have been causing issues.
Right, I guess only having write to the repo is fine as a permission, as long as you can push some code.
What is the worst that could happen btw? With a leaked token? People that could already exfiltrate the token by creating an action already have write access to the org

**Pull request merged**